docs/common/craft-parts/reference/plugins/dotnet_v2_plugin.rst

```diff
@@ -0,0 +1,231 @@
++------------------------------+------------------------+
+
+dotnet-verbosity
+^^^^^^^^^^^^^^^^
+**Type:** string
+
+**Default:** ``"normal"``
+
+Sets the MSBuild log output verbosity for the build. Possible values are:
+``q[uiet]``, ``m[inimal]``, ``n[ormal]``, ``d[etailed]``, and ``diag[nostic]``.
+
+.. _dotnet_version:
+
+dotnet-version
+^^^^^^^^^^^^^^
+**Type:** string
+
+**Default:** ``None``
+
+Sets the .NET version to build the project with. By setting this property, the plugin
+will download the necessary .NET SDK content snap and use it to build the application.
+
+See the :ref:`dotnet-v2-details-begin` section for a more detailed explanation of this property.
+
+.. _restore_flags:
+
+Restore Flags
+~~~~~~~~~~~~~
+
+dotnet-restore-configfile
+^^^^^^^^^^^^^^^^^^^^^^^^^
+**Type:** string
+
+**Default:** ``None``
+
+A path to the NuGet configuration file (nuget.config) to use. If specified, only the
+settings from this file will be used. If not specified, the hierarchy of configuration
+files from the current directory will be used. For more information, see
+`Common NuGet Configurations`_.
+
+dotnet-restore-properties
+^^^^^^^^^^^^^^^^^^^^^^^^^
+**Type:** dict of strings to strings
+
+**Default:** Empty
+
+A list of MSBuild properties to be appended to the restore command in the format of
+``-p:<Key>=<Value>``.
+
+dotnet-restore-sources
+^^^^^^^^^^^^^^^^^^^^^^
+**Type:** list of strings
+
+**Default:** Empty
+
+Specifies the URIs of the NuGet package sources to use during the restore operation.
+This setting overrides all of the sources specified in the *nuget.config* files.
+
+.. _build_flags:
+
+Build Flags
+~~~~~~~~~~~
+
+dotnet-build-framework
+^^^^^^^^^^^^^^^^^^^^^^
+
+**Type:** string
+
+**Default:** ``None``
+
+Compiles for a specific `framework`_. The framework must be defined in the `project file`_.
+Examples: ``net7.0``, ``net462``.
+
+dotnet-build-properties
+^^^^^^^^^^^^^^^^^^^^^^^^^
+**Type:** dict of strings to strings
+
+**Default:** Empty
+
+A list of MSBuild properties to be appended to the build command in the format of
+``-p:<Key>=<Value>``.
+
+.. _publish_flags:
+
+Publish Flags
+~~~~~~~~~~~~~
+
+dotnet-publish-properties
+^^^^^^^^^^^^^^^^^^^^^^^^^
+**Type:** dict of strings to strings
+
+**Default:** Empty
+
+A list of MSBuild properties to be appended to the publish command in the format of
+``-p:<Key>=<Value>``.
+
+.. _dotnet-v2-details-begin:
+
+Dependencies
+------------
+
+The .NET plugin needs the ``dotnet`` executable to build programs. The plugin will
+provision it by itself if :ref:`dotnet_version` is set.
+
+If not, some common means of providing ``dotnet`` are:
+
+* The ``dotnet8`` Ubuntu package, declared as a ``build-package``.
+* The ``dotnet-sdk-80`` snap, declared as a ``build-snap`` from the desired channel.
+
+Another alternative is to define another part with the name ``dotnet-deps``, and
+declare that the part using the ``dotnet`` plugin comes :ref:`after <after>` the
```

Comment:
```suggestion
declare that the part using the .NET plugin (v2) comes :ref:`after <after>` the
```

We should insist on a plain-English name for the plugin, like we have in the title.